src/Valleysoft.DockerCredsProvider/CredsProvider.cs

```diff
@@ -98,7 +99,7 @@ private static async Task<ICredStore> GetCredStoreAsync(string registry, IFileSy
 
             if (configDoc.RootElement.TryGetProperty("auths", out JsonElement authsElement))
             {
-                JsonProperty property = authsElement.EnumerateObject().FirstOrDefault(prop => prop.Name == registry);
+                JsonProperty property = authsElement.EnumerateObject().FirstOrDefault(prop => ConvertToHostname(prop.Name).Equals(ConvertToHostname(registry), StringComparison.InvariantCultureIgnoreCase));
```

Comment:
Since this gets executed in a loop, it'd be better to execute `ConvertToHostname(registry)` only once outside of the loop.

Reply:
done.